src/lib/parse_pdbqt.cpp

```diff
@@ -447,120 +463,101 @@ void postprocess_residue(non_rigid_parsed& nr, parsing_struct& p, context& c) {
-        throw stream_parse_error(count, e.reason);
-    }
+    if(p.atoms.empty()) 
+        throw pdbqt_parse_error("No atoms in this ligand.");
+    if(!torsdof)
+        throw pdbqt_parse_error("Missing TORSDOF keyword.");
+
+    postprocess_ligand(nr, p, c, unsigned(torsdof.get())); // bizarre size_t -> unsigned compiler complaint
 
     VINA_CHECK(nr.atoms_atoms_bonds.dim() == nr.atoms.size());
 }
 
 void parse_pdbqt_ligand(const path& name, non_rigid_parsed& nr, context& c) {
     ifile in(name);
-    unsigned count = 0;
     parsing_struct p;
     boost::optional<unsigned> torsdof;
-    try {
-        parse_pdbqt_aux(in, count, p, c, torsdof, false);
-        if(p.atoms.empty()) 
-            throw parse_error(name, count, "No atoms in the ligand");
-        if(!torsdof)
-            throw parse_error(name, count, "Missing TORSDOF");
-        postprocess_ligand(nr, p, c, unsigned(torsdof.get())); // bizarre size_t -> unsigned compiler complaint
-    }
-    catch(stream_parse_error& e) {
-        throw e.to_parse_error(name);
-    }
+
+    parse_pdbqt_aux(in, p, c, torsdof, false);
+
+    if(p.atoms.empty()) 
+        throw pdbqt_parse_error("No atoms in this ligand.");
+    if(!torsdof)
+        throw pdbqt_parse_error("Missing TORSDOF keyword in this ligand.");
+
+    postprocess_ligand(nr, p, c, unsigned(torsdof.get())); // bizarre size_t -> unsigned compiler complaint
+
     VINA_CHECK(nr.atoms_atoms_bonds.dim() == nr.atoms.size());
 }
 
-void parse_pdbqt_residue(std::istream& in, unsigned& count, parsing_struct& p, context& c) { 
+void parse_pdbqt_residue(std::istream& in, parsing_struct& p, context& c) { 
     boost::optional<unsigned> dummy;
-    parse_pdbqt_aux(in, count, p, c, dummy, true);
+    parse_pdbqt_aux(in, p, c, dummy, true);
 }
 
 void parse_pdbqt_flex(const path& name, non_rigid_parsed& nr, context& c) {
     ifile in(name);
-    unsigned count = 0;
     std::string str;
 
     while(std::getline(in, str)) {
         add_context(c, str);
-        ++count;
+
         if(str.empty()) {} // ignore ""
         else if(starts_with(str, "WARNING")) {} // ignore - AutoDockTools bug workaround
         else if(starts_with(str, "REMARK")) {} // ignore
         else if(starts_with(str, "BEGIN_RES")) {
-            try {
-                parsing_struct p;
-                parse_pdbqt_residue(in, count, p, c);
-                postprocess_residue(nr, p, c);
-            }
-            catch(stream_parse_error& e) {
-                throw e.to_parse_error(name);
-            }
+            parsing_struct p;
+            parse_pdbqt_residue(in, p, c);
+            postprocess_residue(nr, p, c);
         }
         else if(starts_with(str, "MODEL"))
-            throw stream_parse_error(count, "Unexpected multi-MODEL input. Use \"vina_split\" first?");
-        else throw parse_error(name, count, "Unknown or inappropriate tag");
+            throw pdbqt_parse_error("Unexpected multi-MODEL tag found in flex residue.");
```

Comment:
In line 355 you kept `Use \"vina_split\" first.` - maybe have it output here too?

Reply:
I removed it because it concerns only the rigid receptor, and `vina_split` is only for flex residues and ligands (for now).